apis/network/v1alpha3/zone_types.go

```diff
@@ -0,0 +1,133 @@
+	AtProvider                     ZoneObservation `json:"atProvider"`
+}
+
+// ZoneParameters define the desired state of an AWS Route53 Hosted Zone.
+type ZoneParameters struct {
+	// CallerReference is an unique string that identifies the request and that
+	// allows failed Zone create requests to be retried without the risk of
+	// executing the operation twice.
+	// +immutable
+	CallerReference *string `json:"callerref"`
+
+	// Any comments that you want to include about the hosted zone.
+	Comment *string `json:"comment,omitempty"`
+
+	// A value that indicates whether this is a private hosted zone.
+	// +immutable
+	PrivateZone *bool `json:"privatezone,omitempty"`
+
+	// The name of the domain. Specify a fully qualified domain name, for example,
+	// www.example.com. The trailing dot is optional; Amazon Route 53 assumes that
+	// the domain name is fully qualified. This means that Route 53 treats www.example.com
+	// (without a trailing dot) and www.example.com. (with a trailing dot) as identical.
+	//
+	// If you're creating a public hosted zone, this is the name you have registered
+	// with your DNS registrar. If your domain name is registered with a registrar
+	// other than Route 53, change the name servers for your domain to the set of
+	// NameServers that CreateHostedZone returns in DelegationSet.
+	// +immutable
+	Name *string `json:"name"`
+
+	// (Private hosted zones only) The ID of an Amazon VPC that you're
+	// associating with this hosted zone. You can specify only one Amazon VPC
+	// when you create a private hosted zone.
+	// +immutable
+	VPCId *string `json:"VPCId,omitempty"`
+
+	// (Private hosted zones only) The region that an Amazon VPC was created in.
+	// +immutable
+	VPCRegion *string `json:"VPCRegion,omitempty"`
+}
+
+// ZoneObservation keeps the state for the external resource.
+type ZoneObservation struct {
+	// ID is the unique URL representing the new hosted zone.
+	ID string `json:"Id"`
+
+	// The number of resource record sets in the hosted zone.
+	ResourceRecordCount int64 `json:"ResourceRecordSetCount"`
+
+	// Location represents the unique URL of the new hosted zone.
+	Location string `json:"Location"`
+}
+
+// DelegationSet describes the name servers for this hosted zone.
+type DelegationSet struct {
+	// NameServers contains a list of the authoritative name servers for a hosted zone.
+	NameServers []string `json:"NameServers"`
+}
+
+// +kubebuilder:object:root=true
+
+// ZoneList contains a list of Zone.
+type ZoneList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []Zone `json:"items"`
+}
+
+// Update the status
+func (status *ZoneObservation) Update(op *route53.CreateHostedZoneOutput) {
+	if op.HostedZone.Id != nil {
+		status.ID = *op.HostedZone.Id
+	}
+	if op.HostedZone.ResourceRecordSetCount != nil {
+		status.ResourceRecordCount = *op.HostedZone.ResourceRecordSetCount
+	}
+	if op.Location != nil {
+		status.Location = *op.Location
+	}
+}
```

Comment:
We typically prefer to keep these "translation" functions in `pkg/client` rather than as methods on the types.

Reply:
This method has been moved from types file to the client package